src/history/hs_rec.c

```diff
@@ -1010,19 +937,54 @@ static int
 __hs_delete_key_from_pos(
```

Comment:
It should be delete and then reinsert?

Reply:
I don't think so per the reasons discussed in the meeting, please resolve this conversation if that makes sense.

Reply:
Then should we rename is as __hs_reinsert_key_from_pos or __hs_fixup_key_from_pos

Reply:
Updated, thanks.